pandas/tests/dtypes/test_inference.py

```diff
@@ -891,6 +891,19 @@ def test_infer_dtype_period(self):
         arr = np.array([Period("2011-01", freq="D"), Period("2011-02", freq="M")])
         assert lib.infer_dtype(arr, skipna=True) == "period"
 
```

Comment:
do we have a test that infers Interval in a similar way (or can paramterize this one)?

Reply:
As mentioned before (https://github.com/pandas-dev/pandas/pull/37367#issuecomment-762145423), this is already implemented and tested before this PR by #27653